### PR TITLE
GB-3478: Add the equivalent yarn and pnpm support in the CLI

### PR DESCRIPTION
**Pull request by jakubadamw**
# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.

# Type of change

- [ ] 💔 Breaking
- [X] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: cli/crates/server/src/custom_resolvers.rs [hunk 1]
```diff
@@ -113,42 +152,108 @@ async fn build_resolver(
...
+            package_manager,
+            &arguments,
+            resolver_build_artifact_directory_path,
+            tracing,
+            &[],
+        )
+        .await?;
+    }
+
+    // FIXME: This is probably rather fragile. Need to re-check why the wrangler build isn't propagating search paths properly.
```
Comment (yoav-lavi) [inline cli/crates/server/src/custom_resolvers.rs:243]: Could you elaborate on this point please?

**Review by yoav-lavi — APPROVED** [review]
Looks good!

File: cli/crates/server/src/custom_resolvers.rs [hunk 2]
```diff
@@ -13,53 +13,91 @@ use crate::errors::ServerError;
...
+    ("yarn.lock", JavaScriptPackageManager::Yarn),
+];
+
+async fn guess_package_manager_from_package_root(path: impl AsRef<Path>) -> Option<JavaScriptPackageManager> {
+    let package_root = path.as_ref();
+
+    futures_util::future::join_all(LOCK_FILES.iter().map(|(file_name, package_manager)| {
+        let path_to_check = package_root.join(file_name);
+        async move {
+            let file_exists = tokio::fs::try_exists(&path_to_check).await.ok().unwrap_or_default();
```
[the remark this answers is not in the document]
Comment (jakubadamw) [reply cli/crates/server/src/custom_resolvers.rs:87]: IMHO this is impossible, actually, but sure, will revisit.

[the remark this answers is not in the document]
Comment (yoav-lavi) [reply cli/crates/server/src/custom_resolvers.rs:87]: WDYM? It should be `Ok(true)`, `Ok(false)` and `Err(error)`, the third being an IO error (likely permissions)

[the remark this answers is not in the document]
Comment (jakubadamw) [reply cli/crates/server/src/custom_resolvers.rs:88]: Yeah, sorry, I meant to say that this particular failure scenario is unlikely to happen

**Pull request merged**